### PR TITLE
Updates webots.py to search for roles in sub directories

### PR DESCRIPTION
**Pull request by Crafty15**
- webots and fake roles could no longer be found due be being moved to sub-directories.
- This requires that the full path to the role be used. For example: `./b webots build  roles/webots/robocup.role
`
- This [PR](https://github.com/NUbots/NUbook/pull/243) updates NUbook, but it needs updating for the full filepath

### CODE REVIEW

**Review by ysims — APPROVED** [review]
Lgtm

File: tools/webots.py [hunk 1]
```diff
@@ -97,10 +97,12 @@ def register(command):
...
 def get_cmake_flags(roles_to_build):
-    roles_dir = os.path.join(b.project_dir, "roles")
-    roles_glob = os.path.join(roles_dir, "*.role")
 
-    available_roles = [Path(role_path).stem for role_path in glob.glob(roles_glob)]
+    available_roles = []
+    for dirpath, _, fnames in os.walk("roles"):  # iterate over the roles directory and all subdirectories
+        for f in fnames:  # iterate over all the files in the directory
+            if f.endswith(".role"):  # check that the file is a role file
+                available_roles.append(os.path.join(dirpath, f))  # add the role file to our list
```
Comment (CMurtagh-LGTM) [inline tools/webots.py:105]:
```suggestion
    for dirpath, _, fnames in os.walk("roles"):
        for f in fnames:
            if f.endswith(".role"):
                available_roles.append(os.path.join(dirpath, f))
```
Way too many comments.

**Review by CMurtagh-LGTM — CHANGES_REQUESTED** [review]

File: tools/webots.py (hunk 1 above)
Comment (TrentHouliston) [reply tools/webots.py:105]: All pretty self-documenting too!

Comment (Bidski) [reply tools/webots.py:105]: probably more simply achieved with a glob
```
from glob import glob

# ....

# Find all available roles
available_roles = glob(os.path.join("roles", "**", "*.role"), recursive=True)
```

Comment (Crafty15) [reply tools/webots.py]: Brilliant! Thanks

File: tools/webots.py [hunk 2]
```diff
@@ -98,11 +98,8 @@ def register(command):
 
 def get_cmake_flags(roles_to_build):
 
-    available_roles = []
-    for dirpath, _, fnames in os.walk("roles"):  # iterate over the roles directory and all subdirectories
-        for f in fnames:  # iterate over all the files in the directory
-            if f.endswith(".role"):  # check that the file is a role file
-                available_roles.append(os.path.join(dirpath, f))  # add the role file to our list
+    # Find all available roles
+    available_roles = available_roles = glob(os.path.join("roles", "**", "*.role"), recursive=True)
```
Comment (CMurtagh-LGTM) [inline tools/webots.py]:
```suggestion
    available_roles = glob(os.path.join("roles", "**", "*.role"), recursive=True)
```
:stuck_out_tongue:

Also do we need to strip the `roles/` and `.role` from the filename https://github.com/NUbots/NUbots/pull/1010#discussion_r1068685321?

**Review by CMurtagh-LGTM — CHANGES_REQUESTED** [review]

File: tools/webots.py (hunk 2 above)
Comment (Bidski) [reply tools/webots.py]: Don't assume that the b script is always running from the project root directory, while that is commonly the case it doesn't have to be.

`available_roles = glob(os.path.join(b.project_dir, "roles", "**", "*.role"), recursive=True)`

Comment (Crafty15) [reply tools/webots.py]: `available_roles = available_roles = glob(os.path.join("roles", "**", "*.role"), recursive=True)`
I already fixed this.

Comment (Crafty15) [reply tools/webots.py]: > Don't assume that the b script is always running from the project root directory, while that is commonly the case it doesn't have to be.
> 
> `available_roles = glob(os.path.join(b.project_dir, "roles", "**", "*.role"), recursive=True)`

I'll add this when i can test it

Comment (Crafty15) [reply tools/webots.py]: 'twerks

**Review by CMurtagh-LGTM — APPROVED** [review]

**Pull request merged**